src/common/src/hash/key.rs

```diff
@@ -214,10 +214,13 @@ impl Hash for SerializedKey {
     }
 }
 
-/// A special hasher designed our hashmap, which just stores precomputed hash key.
+/// A special hasher designed for [`HashKey`], which stores a hash key from `HashKey::hash()` and
+/// outputs it on `finish()`.
 ///
 /// We need this because we compute hash keys in vectorized fashion, and we store them in this
 /// hasher.
+///
+/// WARN: This should ONLY be used along with [`HashKey`].
 #[derive(Default)]
 pub struct PrecomputedHasher {
```

Comment:
Is it possible to remove this `pub` modifier and exposes JoinHashMap only?

Reply:
Yeah I had thought about it but it's actually used by multiple join executors in both stream/batc